hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/Converter.java

```diff
@@ -318,6 +318,20 @@ protected final QueryException cannotConvert(QueryDataTypeFamily source, QueryDa
         return QueryException.error(SqlErrorCode.DATA_EXCEPTION, message);
     }
 
+    protected final QueryException numericOverflow(QueryDataTypeFamily target, Object val) {
+        return numericOverflow(typeFamily, target, val);
+    }
+
+    protected final QueryException numericOverflow(QueryDataTypeFamily source, QueryDataTypeFamily target, Object val) {
+        String message = "Numeric overflow while converting " + source + " to " + target;
+
+        if (val != null) {
+            message += ": " + val;
```

Comment:
The comment relates to this piece of code and other code places where we dump the value. This could be a potential security vulnerability: we add the value to the message, and it could be printed to the application log. 

Imagine an application where there are two roles - administrator and HR. An HR submits an invalid query that prints something like "Cannot convert VARCHAR to DECIMAL: <salary>". Now, by knowing the nature of the application the administrator can infer sensitive information from logs.

Or imagine that in future we added some sort of row-level security, that allows users to query only rows that satisfy a certain predicate. Again, by carefully constructing the query a user may dump values that he should not have access to.

Note that some values a harmless, e.g. literals. While others could be sensitive, e.g. column values.

@kwart What do you think about that? I think we should avoid printing arbitrary internal values to logs.

Reply:
fixed in 9082f7d2b8af4d33049b70620c22f03e03150206